### PR TITLE
feat: add more field on tx

### PR DESCRIPTION
**Pull request by shaojunda**
1. expose `witnesses`， `cell_deps`, `header_deps` on tx.
2. save output data size.
3. limit output download.
4. save cell occupied capacity.

### CODE REVIEW

File: app/controllers/api/v1/cell_output_data_controller.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class CellOutputDataController < ApplicationController
       def show
         cell_output = CellOutput.where(id: params[:id]).take!
 
+        raise Api::V1::Exceptions::CellOutputDataSizeExceedsLimitError if cell_output&.data_size.to_i > CellOutput::MAXIMUM_DOWNLOADABLE_SIZE
```
Comment (classicalliu) [inline app/controllers/api/v1/cell_output_data_controller.rb]: If `cell_output` is null, will `>` raise an error ?

Comment (shaojunda) [reply app/controllers/api/v1/cell_output_data_controller.rb]: will not raise error, because `nil.to_i == 0`

Comment (classicalliu) [reply app/controllers/api/v1/cell_output_data_controller.rb]: Will not call `to_i` if `cell_output` is null

Comment (shaojunda) [reply app/controllers/api/v1/cell_output_data_controller.rb]: image
why?

Comment (classicalliu) [reply app/controllers/api/v1/cell_output_data_controller.rb]: Ohhhh I see, no `&` between `data_size` and `to_i`

**Pull request merged**